From d6a1ca122a7454d461cd594e25a75b52c11bd148 Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Mon, 4 Jul 2011 19:12:01 +0800
Subject: [PATCH 42/45] check ha before doing comparison

---
 xdiff/xhistogram.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/xdiff/xhistogram.c b/xdiff/xhistogram.c
index ca19a91..07b3dd7 100644
--- a/xdiff/xhistogram.c
+++ b/xdiff/xhistogram.c
@@ -43,7 +43,8 @@ static int cmp_env(xpparam_t const *xpp, xdfenv_t *env,
 {
 	xrecord_t *r1 = get_rec(env, side1, line1),
 		  *r2 = get_rec(env, side2, line2);
-	return xdl_recmatch(r1->ptr, r1->size, r2->ptr, r2->size,
+	return r1->ha == r2->ha &&
+		xdl_recmatch(r1->ptr, r1->size, r2->ptr, r2->size,
 			    xpp->flags);
 }
 
-- 
1.7.6.12.g6486a.dirty

