From 5bc0d462be438c04b89204e38691b264a96157fe Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Sun, 31 Jul 2011 21:02:20 +0800
Subject: [PATCH 009/130] series: reorder

---
 guilt/gsoc-diff/series |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/guilt/gsoc-diff/series b/guilt/gsoc-diff/series
index 3d7ad52..6b6c1e6 100644
--- a/guilt/gsoc-diff/series
+++ b/guilt/gsoc-diff/series
@@ -1,2 +1,2 @@
-remove-mmfile_next
 use-memset
+remove-mmfile_next
-- 
1.7.4.msysgit.0

