From dbd1ca493d5f354aced2c7eb9029580f7a577d10 Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Sun, 31 Jul 2011 21:44:45 +0800
Subject: [PATCH 025/130] MAX_CNT: reuse value

---
 guilt/gsoc-diff/teach-histogram |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/guilt/gsoc-diff/teach-histogram b/guilt/gsoc-diff/teach-histogram
index 8fed7b4..4d079a1 100644
--- a/guilt/gsoc-diff/teach-histogram
+++ b/guilt/gsoc-diff/teach-histogram
@@ -95,7 +95,7 @@ index ad033a8..7a92ea9 100644
  #endif /* #if !defined(XDIFFI_H) */
 diff --git a/xdiff/xhistogram.c b/xdiff/xhistogram.c
 new file mode 100644
-index 0000000..d199cb7
+index 0000000..679a2df
 --- /dev/null
 +++ b/xdiff/xhistogram.c
 @@ -0,0 +1,338 @@
@@ -108,7 +108,7 @@ index 0000000..d199cb7
 +#define REC_PTR_MASK	((1 << 28) - 1)
 +#define REC_CNT_MASK	((1 << 8) - 1)
 +#define MAX_PTR		REC_PTR_MASK
-+#define MAX_CNT		(1 << 8) - 1
++#define MAX_CNT		REC_CNT_MASK
 +
 +#define REC_CREATE(n, p, c) \
 +	(((unsigned long) n << REC_NEXT_SHIFT) \
-- 
1.7.4.msysgit.0

