From e03487a13312d68b61b591e5054cb70ea0a571a8 Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Sun, 31 Jul 2011 21:45:49 +0800
Subject: [PATCH 032/130] remove useless arguments

---
 guilt/gsoc-diff/teach-histogram |   11 +++++------
 1 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/guilt/gsoc-diff/teach-histogram b/guilt/gsoc-diff/teach-histogram
index c245d36..7b12425 100644
--- a/guilt/gsoc-diff/teach-histogram
+++ b/guilt/gsoc-diff/teach-histogram
@@ -95,10 +95,10 @@ index ad033a8..7a92ea9 100644
  #endif /* #if !defined(XDIFFI_H) */
 diff --git a/xdiff/xhistogram.c b/xdiff/xhistogram.c
 new file mode 100644
-index 0000000..58d78a0
+index 0000000..636550d
 --- /dev/null
 +++ b/xdiff/xhistogram.c
-@@ -0,0 +1,341 @@
+@@ -0,0 +1,340 @@
 +#include "xinclude.h"
 +#include "xtypes.h"
 +#include "xdiff.h"
@@ -337,8 +337,7 @@ index 0000000..58d78a0
 +				  line1, count1, line2, count2);
 +}
 +
-+static int histogram_diff(mmfile_t *file1, mmfile_t *file2,
-+	xpparam_t const *xpp, xdfenv_t *env,
++static int histogram_diff(xpparam_t const *xpp, xdfenv_t *env,
 +	int line1, int count1, int line2, int count2)
 +{
 +	struct histindex index;
@@ -412,10 +411,10 @@ index 0000000..58d78a0
 +			for (ptr = 0; ptr < count2; ptr++)
 +				env->xdf2.rchg[line2 + ptr - 1] = 1;
 +		} else {
-+			result = histogram_diff(file1, file2, xpp, env,
++			result = histogram_diff(xpp, env,
 +				line1, lcs.begin1 - line1,
 +				line2, lcs.begin2 - line2);
-+			result = histogram_diff(file1, file2, xpp, env,
++			result = histogram_diff(xpp, env,
 +				lcs.end1 + 1, LINE_END(1) - lcs.end1,
 +				lcs.end2 + 1, LINE_END(2) - lcs.end2);
 +			result *= -1;
-- 
1.7.4.msysgit.0

