From 4f63e6c9bcd879f08c89a69b8151079443aea6f5 Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Sun, 31 Jul 2011 21:48:00 +0800
Subject: [PATCH 051/130] remove unused member, previous

---
 guilt/gsoc-diff/teach-histogram |   10 ++++------
 1 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/guilt/gsoc-diff/teach-histogram b/guilt/gsoc-diff/teach-histogram
index fe3e99c..23f7348 100644
--- a/guilt/gsoc-diff/teach-histogram
+++ b/guilt/gsoc-diff/teach-histogram
@@ -95,10 +95,10 @@ index ad033a8..7a92ea9 100644
  #endif /* #if !defined(XDIFFI_H) */
 diff --git a/xdiff/xhistogram.c b/xdiff/xhistogram.c
 new file mode 100644
-index 0000000..915b080
+index 0000000..c9fd76e
 --- /dev/null
 +++ b/xdiff/xhistogram.c
-@@ -0,0 +1,335 @@
+@@ -0,0 +1,333 @@
 +#include "xinclude.h"
 +#include "xtypes.h"
 +#include "xdiff.h"
@@ -112,7 +112,7 @@ index 0000000..915b080
 +struct histindex {
 +	struct record {
 +		unsigned int ptr, cnt;
-+		struct record *next, *previous;
++		struct record *next;
 +	} **records, /* an ocurrence */
 +	  **line_map; /* map of line to record chain */
 +	unsigned int table_bits,
@@ -182,9 +182,7 @@ index 0000000..915b080
 +					new_cnt = MAX_CNT;
 +				new_rec->cnt = new_cnt;
 +
-+				new_rec->previous = NULL;
 +				new_rec->next = rec;
-+				rec->previous = new_rec;
 +				*rec_chain = new_rec;
 +				LINE_MAP(index, ptr) = new_rec;
 +				goto continue_scan;
@@ -203,7 +201,7 @@ index 0000000..915b080
 +			return -1;
 +		new_rec->ptr = ptr;
 +		new_rec->cnt = 1;
-+		new_rec->next = new_rec->previous = NULL;
++		new_rec->next = NULL;
 +		LINE_MAP(index, ptr) = new_rec;
 +		*rec_chain = new_rec;
 +
-- 
1.7.4.msysgit.0

