From 9dc526869ca75a110cad5ca38e0823ce016057b6 Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Sun, 31 Jul 2011 21:48:18 +0800
Subject: [PATCH 055/130] check xdl_cha_init() status

---
 guilt/gsoc-diff/teach-histogram |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/guilt/gsoc-diff/teach-histogram b/guilt/gsoc-diff/teach-histogram
index bfdcdf3..2027e69 100644
--- a/guilt/gsoc-diff/teach-histogram
+++ b/guilt/gsoc-diff/teach-histogram
@@ -95,7 +95,7 @@ index ad033a8..7a92ea9 100644
  #endif /* #if !defined(XDIFFI_H) */
 diff --git a/xdiff/xhistogram.c b/xdiff/xhistogram.c
 new file mode 100644
-index 0000000..4542d7f
+index 0000000..635eacd
 --- /dev/null
 +++ b/xdiff/xhistogram.c
 @@ -0,0 +1,332 @@
@@ -378,7 +378,7 @@ index 0000000..4542d7f
 +	/* so that xdl_cha_free() doesn't break */
 +	index.rcha.head = NULL;
 +	/* lines / 4 + 1 comes from xprepare.c:xdl_prepare_ctx() */
-+	if (!xdl_cha_init(&index.rcha, sizeof(struct record), count1 / 4 + 1))
++	if (xdl_cha_init(&index.rcha, sizeof(struct record), count1 / 4 + 1) < 0)
 +		goto cleanup;
 +
 +	index.ptr_shift = line1;
-- 
1.7.4.msysgit.0

