From 7ab6ed54ac66572366af5435dacb959680654adb Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Sun, 31 Jul 2011 23:56:45 +0800
Subject: [PATCH 096/130] xdl_prepare_ctx: simplify cur advancement

---
 guilt/gsoc-diff/trim-ends |   19 +++++++++----------
 1 files changed, 9 insertions(+), 10 deletions(-)

diff --git a/guilt/gsoc-diff/trim-ends b/guilt/gsoc-diff/trim-ends
index f3312ab..2e0daed 100644
--- a/guilt/gsoc-diff/trim-ends
+++ b/guilt/gsoc-diff/trim-ends
@@ -1,5 +1,5 @@
 diff --git a/xdiff/xprepare.c b/xdiff/xprepare.c
-index 3ebad0f..94cad7f 100644
+index 3ebad0f..9178626 100644
 --- a/xdiff/xprepare.c
 +++ b/xdiff/xprepare.c
 @@ -53,11 +53,12 @@ static void xdl_free_classifier(xdlclassifier_t *cf);
@@ -58,16 +58,15 @@ index 3ebad0f..94cad7f 100644
  	unsigned int hbits;
  	long nrec, hsize, bsize;
  	unsigned long hav;
-@@ -164,7 +196,15 @@ static int xdl_prepare_ctx(mmfile_t *mf, long narec, xpparam_t const *xpp,
+@@ -164,7 +196,14 @@ static int xdl_prepare_ctx(mmfile_t *mf, long narec, xpparam_t const *xpp,
  	if ((cur = blk = xdl_mmfile_first(mf, &bsize)) != NULL) {
  		for (top = blk + bsize; cur < top; ) {
  			prev = cur;
 -			hav = xdl_hash_record(&cur, top, xpp->flags);
 +			if (cur < xdf->rstart) {
-+				if (arec) {
-+					cur += arec[1]->ptr - arec[0]->ptr;
-+					arec++;
-+				} else
++				if (arec)
++					cur += (arec++)[0]->size;
++				else
 +					cur = memchr(cur, '\n', top - cur) + 1;
 +				hav = 0;
 +			} else
@@ -75,7 +74,7 @@ index 3ebad0f..94cad7f 100644
  			if (nrec >= narec) {
  				narec *= 2;
  				if (!(rrecs = (xrecord_t **) xdl_realloc(recs, narec * sizeof(xrecord_t *))))
-@@ -178,7 +218,7 @@ static int xdl_prepare_ctx(mmfile_t *mf, long narec, xpparam_t const *xpp,
+@@ -178,7 +217,7 @@ static int xdl_prepare_ctx(mmfile_t *mf, long narec, xpparam_t const *xpp,
  			crec->ha = hav;
  			recs[nrec++] = crec;
  
@@ -84,7 +83,7 @@ index 3ebad0f..94cad7f 100644
  				goto abort;
  		}
  	}
-@@ -200,7 +240,6 @@ static int xdl_prepare_ctx(mmfile_t *mf, long narec, xpparam_t const *xpp,
+@@ -200,7 +239,6 @@ static int xdl_prepare_ctx(mmfile_t *mf, long narec, xpparam_t const *xpp,
  	xdf->rindex = rindex;
  	xdf->nreff = 0;
  	xdf->ha = ha;
@@ -92,7 +91,7 @@ index 3ebad0f..94cad7f 100644
  	xdf->dend = nrec - 1;
  
  	return 0;
-@@ -240,12 +279,18 @@ int xdl_prepare_env(mmfile_t *mf1, mmfile_t *mf2, xpparam_t const *xpp,
+@@ -240,12 +278,18 @@ int xdl_prepare_env(mmfile_t *mf1, mmfile_t *mf2, xpparam_t const *xpp,
  		return -1;
  	}
  
@@ -113,7 +112,7 @@ index 3ebad0f..94cad7f 100644
  
  		xdl_free_ctx(&xe->xdf1);
  		xdl_free_classifier(&cf);
-@@ -403,39 +448,9 @@ static int xdl_cleanup_records(xdfile_t *xdf1, xdfile_t *xdf2) {
+@@ -403,39 +447,9 @@ static int xdl_cleanup_records(xdfile_t *xdf1, xdfile_t *xdf2) {
  }
  
  
-- 
1.7.4.msysgit.0

