From 90dc2e75eed0a70b744c43e056fbeeeb68c0b2e5 Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Sun, 31 Jul 2011 21:43:50 +0800
Subject: [PATCH 022/130] simplify lcs check

---
 guilt/gsoc-diff/teach-histogram |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/guilt/gsoc-diff/teach-histogram b/guilt/gsoc-diff/teach-histogram
index 5cae7d2..87854f4 100644
--- a/guilt/gsoc-diff/teach-histogram
+++ b/guilt/gsoc-diff/teach-histogram
@@ -95,7 +95,7 @@ index ad033a8..7a92ea9 100644
  #endif /* #if !defined(XDIFFI_H) */
 diff --git a/xdiff/xhistogram.c b/xdiff/xhistogram.c
 new file mode 100644
-index 0000000..5374ff1
+index 0000000..55e2d17
 --- /dev/null
 +++ b/xdiff/xhistogram.c
 @@ -0,0 +1,338 @@
@@ -403,7 +403,7 @@ index 0000000..5374ff1
 +		result = fall_back_to_classic_diff(&index, line1, count1, line2, count2);
 +	else {
 +		result = 0;
-+		if (lcs.begin1 == lcs.end1 && lcs.begin2 == lcs.end2) {
++		if (lcs.begin1 == 0 && lcs.begin2 == 0) {
 +			int ptr;
 +			for (ptr = 0; ptr < count1; ptr++)
 +				env->xdf1.rchg[line1 + ptr - 1] = 1;
-- 
1.7.4.msysgit.0

