From 3a22ff143a6ef47cf1085cb071525fd3068f2388 Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Sun, 31 Jul 2011 21:47:20 +0800
Subject: [PATCH 046/130] simplify np operations

---
 guilt/gsoc-diff/teach-histogram |   19 ++++++++++++-------
 1 files changed, 12 insertions(+), 7 deletions(-)

diff --git a/guilt/gsoc-diff/teach-histogram b/guilt/gsoc-diff/teach-histogram
index 9870a32..652ad00 100644
--- a/guilt/gsoc-diff/teach-histogram
+++ b/guilt/gsoc-diff/teach-histogram
@@ -95,10 +95,10 @@ index ad033a8..7a92ea9 100644
  #endif /* #if !defined(XDIFFI_H) */
 diff --git a/xdiff/xhistogram.c b/xdiff/xhistogram.c
 new file mode 100644
-index 0000000..d24f666
+index 0000000..f8312fb
 --- /dev/null
 +++ b/xdiff/xhistogram.c
-@@ -0,0 +1,330 @@
+@@ -0,0 +1,335 @@
 +#include "xinclude.h"
 +#include "xtypes.h"
 +#include "xdiff.h"
@@ -220,11 +220,17 @@ index 0000000..d24f666
 +	return 0;
 +}
 +
++static int get_next_ptr(struct histindex *index, int ptr)
++{
++	struct record *rec = INDEX_NEXT(index, ptr).rec->next;
++	return rec ? rec->ptr : 0;
++}
++
 +static int try_lcs(struct histindex *index, struct region *lcs, int b_ptr,
 +	int line1, int count1, int line2, int count2)
 +{
 +	unsigned int b_next = b_ptr + 1;
-+	struct record *rec = index->records[table_hash(index, 2, b_ptr)], *rec_ahead;
++	struct record *rec = index->records[table_hash(index, 2, b_ptr)];
 +	unsigned int as, ae, bs, be, np, rc;
 +	int should_break;
 +
@@ -242,8 +248,7 @@ index 0000000..d24f666
 +		index->has_common = 1;
 +		for (;;) {
 +			should_break = 0;
-+			rec_ahead = INDEX_NEXT(index, as).rec->next;
-+			np = rec_ahead ? rec_ahead->ptr : 0;
++			np = get_next_ptr(index, as);
 +			bs = b_ptr;
 +			ae = as;
 +			be = bs;
@@ -278,8 +283,8 @@ index 0000000..d24f666
 +				break;
 +
 +			while (np <= ae) {
-+				rec_ahead = rec_ahead->next;
-+				if (!rec_ahead || (np = rec_ahead->ptr) == 0) {
++				np = get_next_ptr(index, np);
++				if (np == 0) {
 +					should_break = 1;
 +					break;
 +				}
-- 
1.7.4.msysgit.0

