From ecfcf7ab0702f0377d5b9caaf224335beeb6f79c Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Sun, 31 Jul 2011 21:50:09 +0800
Subject: [PATCH 057/130] add patch text

---
 guilt/gsoc-diff/teach-histogram |   13 +++++++++++++
 1 files changed, 13 insertions(+), 0 deletions(-)

diff --git a/guilt/gsoc-diff/teach-histogram b/guilt/gsoc-diff/teach-histogram
index e5c820e..fcad9ca 100644
--- a/guilt/gsoc-diff/teach-histogram
+++ b/guilt/gsoc-diff/teach-histogram
@@ -1,3 +1,16 @@
+teach --histogram to diff
+
+Port JGit's HistogramDiff algorithm over to C. Rough numbers (TODO) show
+that it is faster than its --patience cousin, as well as the default
+Meyers algorithm.
+
+The implementation has been reworked to use structs and pointers,
+instead of bitmasks, thus doing away with JGit's 2^28 line limit.
+
+We also use xdiff's default hash table implementation (xdl_hash_bits()
+with XDL_HASHLONG()) for convenience.
+
+Signed-off-by: Tay Ray Chuan <rctay89@gmail.com>
 diff --git a/Makefile b/Makefile
 index 775ee83..9930e6b 100644
 --- a/Makefile
-- 
1.7.4.msysgit.0

