From c146285691435882e63e38e19e7131cec147aba2 Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Sun, 31 Jul 2011 22:12:11 +0800
Subject: [PATCH 070/130] patch text

---
 guilt/gsoc-diff/skip-classify |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/guilt/gsoc-diff/skip-classify b/guilt/gsoc-diff/skip-classify
index 33aca1b..8d36954 100644
--- a/guilt/gsoc-diff/skip-classify
+++ b/guilt/gsoc-diff/skip-classify
@@ -2,9 +2,13 @@ xdiff/xprepare: skip classification
 
 xdl_classify_record() replaces the hash member (xrecord_t.ha) with a
 unique identifier of the record/line, such that the hashes are
-"linearized" (as Dscho calls it in xpatience.c).
+"linearized" (as Dscho calls it in xpatience.c). The records are then
+placed into a hash table (xrecord_t.rhash).
 
-We don't want that in histogram diff, so turn it off.
+We don't want that in histogram diff, so we omit calls to record
+classification functions (that is, xdl_init_classifier() and
+xdl_classify_record()). We also skip allocating memory to the hash
+table, rhash, as it is no longer used.
 
 This gives us a small boost in performance.
 
-- 
1.7.4.msysgit.0

