From 178b163e364f8c1fa977a304c763bd81c3e5552a Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Sun, 31 Jul 2011 21:55:39 +0800
Subject: [PATCH 074/130] rely on xdl_trim_ends()

---
 guilt/gsoc-diff/teach-histogram |   35 ++++++-----------------------------
 1 files changed, 6 insertions(+), 29 deletions(-)

diff --git a/guilt/gsoc-diff/teach-histogram b/guilt/gsoc-diff/teach-histogram
index aa76fc6..21507ab 100644
--- a/guilt/gsoc-diff/teach-histogram
+++ b/guilt/gsoc-diff/teach-histogram
@@ -108,10 +108,10 @@ index ad033a8..7a92ea9 100644
  #endif /* #if !defined(XDIFFI_H) */
 diff --git a/xdiff/xhistogram.c b/xdiff/xhistogram.c
 new file mode 100644
-index 0000000..391333a
+index 0000000..cb4f3d3
 --- /dev/null
 +++ b/xdiff/xhistogram.c
-@@ -0,0 +1,384 @@
+@@ -0,0 +1,361 @@
 +/*
 + * Copyright (C) 2010, Google Inc.
 + * and other copyright owners as documented in JGit's IP log.
@@ -216,7 +216,7 @@ index 0000000..391333a
 +	(cmp_recs(xpp, REC(env, s1, l1), REC(env, s2, l2)))
 +
 +#define CMP(i, s1, l1, s2, l2) \
-+	(CMP_ENV(i->xpp, i->env, s1, l1, s2, l2))
++	(cmp_recs(i->xpp, REC(i->env, s1, l1), REC(i->env, s2, l2)))
 +
 +#define TABLE_HASH(index, side, line) \
 +	XDL_HASHLONG((REC(index->env, side, line))->ha, index->table_bits)
@@ -362,23 +362,6 @@ index 0000000..391333a
 +	return index->has_common && index->max_chain_length < index->cnt;
 +}
 +
-+static void reduce_common_start_end(xpparam_t const *xpp, xdfenv_t *env,
-+	int *line1, int *count1, int *line2, int *count2)
-+{
-+	if (*count1 <= 1 || *count2 <= 1)
-+		return;
-+	while (*count1 > 1 && *count2 > 1 && CMP_ENV(xpp, env, 1, *line1, 2, *line2)) {
-+		(*line1)++;
-+		(*count1)--;
-+		(*line2)++;
-+		(*count2)--;
-+	}
-+	while (*count1 > 1 && *count2 > 1 && CMP_ENV(xpp, env, 1, LINE_END_PTR(1), 2, LINE_END_PTR(2))) {
-+		(*count1)--;
-+		(*count2)--;
-+	}
-+}
-+
 +static int fall_back_to_classic_diff(struct histindex *index,
 +		int line1, int count1, int line2, int count2)
 +{
@@ -483,16 +466,10 @@ index 0000000..391333a
 +int xdl_do_histogram_diff(mmfile_t *file1, mmfile_t *file2,
 +	xpparam_t const *xpp, xdfenv_t *env)
 +{
-+	int line1, line2, count1, count2;
-+
 +	if (xdl_prepare_env(file1, file2, xpp, env) < 0)
 +		return -1;
 +
-+	line1 = line2 = 1;
-+	count1 = env->xdf1.nrec;
-+	count2 = env->xdf2.nrec;
-+
-+	reduce_common_start_end(xpp, env, &line1, &count1, &line2, &count2);
-+
-+	return histogram_diff(xpp, env, line1, count1, line2, count2);
++	return histogram_diff(xpp, env,
++		env->xdf1.dstart + 1, env->xdf1.dend - env->xdf1.dstart + 1,
++		env->xdf2.dstart + 1, env->xdf2.dend - env->xdf2.dstart + 1);
 +}
-- 
1.7.4.msysgit.0

