From 4fc8345b4d15151c3bfee2e6aa7d3c8b157da9e0 Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Sun, 31 Jul 2011 23:59:55 +0800
Subject: [PATCH 104/130] xdl_trim_head(): minor space issue

---
 guilt/gsoc-diff/trim-ends |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/guilt/gsoc-diff/trim-ends b/guilt/gsoc-diff/trim-ends
index a764ac8..2e4f20d 100644
--- a/guilt/gsoc-diff/trim-ends
+++ b/guilt/gsoc-diff/trim-ends
@@ -1,5 +1,5 @@
 diff --git a/xdiff/xprepare.c b/xdiff/xprepare.c
-index 3ebad0f..98f5766 100644
+index 3ebad0f..113c672 100644
 --- a/xdiff/xprepare.c
 +++ b/xdiff/xprepare.c
 @@ -26,6 +26,7 @@
@@ -42,7 +42,7 @@ index 3ebad0f..98f5766 100644
  
  
 +static void xdl_trim_head(mmfile_t *mf1, mmfile_t *mf2, xpparam_t const *xpp,
-+			 xdfile_t *xdf1, xdfile_t *xdf2) {
++			  xdfile_t *xdf1, xdfile_t *xdf2) {
 +
 +	long trimmed = 0, recovered = 0;
 +	long smaller = XDL_MIN(mf1->size, mf2->size);
-- 
1.7.4.msysgit.0

