From 92431a54e3661fdf8013a0c7a997ab359244fd3f Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Mon, 1 Aug 2011 00:00:12 +0800
Subject: [PATCH 108/130] set hash to null for common tail

---
 guilt/gsoc-diff/trim-ends |   40 ++++++++++++++++++++++++++++------------
 1 files changed, 28 insertions(+), 12 deletions(-)

diff --git a/guilt/gsoc-diff/trim-ends b/guilt/gsoc-diff/trim-ends
index 2b129fc..a6b50f0 100644
--- a/guilt/gsoc-diff/trim-ends
+++ b/guilt/gsoc-diff/trim-ends
@@ -1,5 +1,5 @@
 diff --git a/xdiff/xprepare.c b/xdiff/xprepare.c
-index 3ebad0f..a93acc0 100644
+index 3ebad0f..e2bf4cf 100644
 --- a/xdiff/xprepare.c
 +++ b/xdiff/xprepare.c
 @@ -26,6 +26,7 @@
@@ -37,7 +37,7 @@ index 3ebad0f..a93acc0 100644
  			break;
  
  	if (!rcrec) {
-@@ -130,8 +132,95 @@ static int xdl_classify_record(xdlclassifier_t *cf, xrecord_t **rhash, unsigned
+@@ -130,8 +132,109 @@ static int xdl_classify_record(xdlclassifier_t *cf, xrecord_t **rhash, unsigned
  }
  
  
@@ -121,9 +121,23 @@ index 3ebad0f..a93acc0 100644
 +}
 +
 +
-+static void xdl_set_dend(xdfile_t *xdf, long lim, long ntail)
++static void xdl_set_dend(xdfile_t *xdf, xpparam_t const *xpp,
++			 long lim, long ntail)
 +{
++	long i, ha;
++	xrecord_t *rec;
++	char const *old_ptr;
++
 +	xdf->dend = xdf->nrec - XDL_MIN(lim - xdf->dstart, ntail) - 1;
++
++	i = xdf->nrec - ntail;
++	while (i < xdf->nrec) {
++		rec = xdf->recs[i];
++		old_ptr = rec->ptr;
++		rec->ha = xdl_hash_record(&rec->ptr, rec->ptr + rec->size, xpp->flags);
++		rec->ptr = old_ptr;
++		i++;
++	}
 +}
 +
 +
@@ -134,7 +148,7 @@ index 3ebad0f..a93acc0 100644
  	unsigned int hbits;
  	long nrec, hsize, bsize;
  	unsigned long hav;
-@@ -161,10 +250,21 @@ static int xdl_prepare_ctx(mmfile_t *mf, long narec, xpparam_t const *xpp,
+@@ -161,10 +264,23 @@ static int xdl_prepare_ctx(mmfile_t *mf, long narec, xpparam_t const *xpp,
  	memset(rhash, 0, hsize * sizeof(xrecord_t *));
  
  	nrec = 0;
@@ -143,8 +157,6 @@ index 3ebad0f..a93acc0 100644
  		for (top = blk + bsize; cur < top; ) {
  			prev = cur;
 -			hav = xdl_hash_record(&cur, top, xpp->flags);
-+			if (cur > xdf->rend)
-+				(*ntail)++;
 +			if (cur < xdf->rstart) {
 +				if (arec)
 +					cur += (arec++)[0]->size;
@@ -152,12 +164,16 @@ index 3ebad0f..a93acc0 100644
 +					cur = memchr(cur, '\n', top - cur) + 1;
 +				hav = 0;
 +				xdf->dstart++;
++			} else if (cur > xdf->rend) {
++				cur = memchr(cur, '\n', top - cur) + 1;
++				hav = 0;
++				(*ntail)++;
 +			} else
 +				hav = xdl_hash_record(&cur, top, xpp->flags);
  			if (nrec >= narec) {
  				narec *= 2;
  				if (!(rrecs = (xrecord_t **) xdl_realloc(recs, narec * sizeof(xrecord_t *))))
-@@ -200,8 +300,6 @@ static int xdl_prepare_ctx(mmfile_t *mf, long narec, xpparam_t const *xpp,
+@@ -200,8 +316,6 @@ static int xdl_prepare_ctx(mmfile_t *mf, long narec, xpparam_t const *xpp,
  	xdf->rindex = rindex;
  	xdf->nreff = 0;
  	xdf->ha = ha;
@@ -166,7 +182,7 @@ index 3ebad0f..a93acc0 100644
  
  	return 0;
  
-@@ -229,7 +327,9 @@ static void xdl_free_ctx(xdfile_t *xdf) {
+@@ -229,7 +343,9 @@ static void xdl_free_ctx(xdfile_t *xdf) {
  
  int xdl_prepare_env(mmfile_t *mf1, mmfile_t *mf2, xpparam_t const *xpp,
  		    xdfenv_t *xe) {
@@ -176,7 +192,7 @@ index 3ebad0f..a93acc0 100644
  	xdlclassifier_t cf;
  
  	enl1 = xdl_guess_lines(mf1) + 1;
-@@ -240,18 +340,25 @@ int xdl_prepare_env(mmfile_t *mf1, mmfile_t *mf2, xpparam_t const *xpp,
+@@ -240,18 +356,25 @@ int xdl_prepare_env(mmfile_t *mf1, mmfile_t *mf2, xpparam_t const *xpp,
  		return -1;
  	}
  
@@ -198,13 +214,13 @@ index 3ebad0f..a93acc0 100644
  	}
  
 +	lim = XDL_MIN(xe->xdf1.nrec, xe->xdf2.nrec);
-+	xdl_set_dend(&xe->xdf1, lim, ntail1);
-+	xdl_set_dend(&xe->xdf2, lim, ntail2);
++	xdl_set_dend(&xe->xdf1, xpp, lim, ntail1);
++	xdl_set_dend(&xe->xdf2, xpp, lim, ntail2);
 +
  	xdl_free_classifier(&cf);
  
  	if (!(xpp->flags & XDF_PATIENCE_DIFF) &&
-@@ -403,39 +510,9 @@ static int xdl_cleanup_records(xdfile_t *xdf1, xdfile_t *xdf2) {
+@@ -403,39 +526,9 @@ static int xdl_cleanup_records(xdfile_t *xdf1, xdfile_t *xdf2) {
  }
  
  
-- 
1.7.4.msysgit.0

