From 3a8e69702c69a3d915874fc649d560501fe06c60 Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Thu, 30 Jun 2011 16:36:26 +0800
Subject: [PATCH 19/28] check if ptr field is within bounds

---
 xdiff/xhistogram.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/xdiff/xhistogram.c b/xdiff/xhistogram.c
index cb57a68..2aa4128 100644
--- a/xdiff/xhistogram.c
+++ b/xdiff/xhistogram.c
@@ -98,6 +98,8 @@ static int scanA(struct histindex *index, int line1, int count1)
 		if (chain_len == index->max_chain_length)
 			return -1;
 
+		if (index->recs_count >= MAX_PTR)
+			return -1;
 		rec_idx = ++index->recs_count;
 		if (rec_idx == index->recs_size) {
 			index->recs_size = XDL_MIN(index->recs_size << 1, 1 + count1);
-- 
1.7.6.12.g6486a.dirty

