From 195de387cbaddb997052149072587725e8210cae Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Mon, 4 Jul 2011 18:40:46 +0800
Subject: [PATCH 24/28] terminate earlier

---
 xdiff/xhistogram.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/xdiff/xhistogram.c b/xdiff/xhistogram.c
index ce8cd9c..4ebc723 100644
--- a/xdiff/xhistogram.c
+++ b/xdiff/xhistogram.c
@@ -245,6 +245,9 @@ static int histogram_diff(struct histindex *index,
 	if (count1 <= 0 && count2 <= 0)
 		return 0;
 
+	if (LINE_END(1) >= MAX_PTR)
+		return -1;
+
 	if (!count1) {
 		while(count2--)
 			index->env->xdf2.rchg[line2++ - 1] = 1;
@@ -265,9 +268,6 @@ static int histogram_diff(struct histindex *index,
 	memset(index->next, 0, index->size * sizeof(int));
 	memset(index->rec_idxs, 0, index->size * sizeof(int));
 
-	if (LINE_END(1) >= MAX_PTR)
-		return -1;
-
 	memset(&lcs, 0, sizeof(lcs));
 	if (find_lcs(index, &lcs, line1, count1, line2, count2))
 		result = fall_back_to_classic_diff(index, line1, count1, line2, count2);
-- 
1.7.6.12.g6486a.dirty

