From d033f9af138754ba03f3a500f507990dd9f246f4 Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Fri, 1 Jul 2011 19:03:47 +0800
Subject: [PATCH 32/41] simplify INDEX_NEXT(index, ptr).rec->cnt

---
 xdiff/xhistogram.c |   10 ++++++++--
 1 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/xdiff/xhistogram.c b/xdiff/xhistogram.c
index 12784a1..c945350 100644
--- a/xdiff/xhistogram.c
+++ b/xdiff/xhistogram.c
@@ -124,6 +124,12 @@ static int get_next_ptr(struct histindex *index, int ptr)
 	return rec ? rec->ptr : 0;
 }
 
+static int get_cnt(struct histindex *index, int ptr)
+{
+	struct record *rec = INDEX_NEXT(index, ptr).rec;
+	return rec ? rec->cnt : 0;
+}
+
 static int try_lcs(struct histindex *index, struct region *lcs, int b_ptr,
 	int line1, int count1, int line2, int count2)
 {
@@ -157,14 +163,14 @@ static int try_lcs(struct histindex *index, struct region *lcs, int b_ptr,
 				as--;
 				bs--;
 				if (1 < rc)
-					rc = XDL_MIN(rc, INDEX_NEXT(index, as).rec->cnt);
+					rc = XDL_MIN(rc, get_cnt(index, as));
 			}
 			while (ae < LINE_END(1) && be < LINE_END(2)
 				&& cmp(index, 1, ae+1, 2, be+1)) {
 				ae++;
 				be++;
 				if (1 < rc)
-					rc = XDL_MIN(rc, INDEX_NEXT(index, ae).rec->cnt);
+					rc = XDL_MIN(rc, get_cnt(index, ae));
 			}
 
 			if (b_next < be)
-- 
1.7.6.12.g6486a.dirty

