From 80a93fc58de976a1c65e777047f4c2c42d47f703 Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Sat, 2 Jul 2011 02:28:21 +0800
Subject: [PATCH 39/41] get_cnt(): fix

---
 xdiff/xhistogram.c |   18 +++++++++++++++---
 1 files changed, 15 insertions(+), 3 deletions(-)

diff --git a/xdiff/xhistogram.c b/xdiff/xhistogram.c
index 23cbb30..986383f 100644
--- a/xdiff/xhistogram.c
+++ b/xdiff/xhistogram.c
@@ -144,9 +144,21 @@ static void get_slice(struct record *rec,
 
 static int get_cnt(struct histindex *index, int line1, int count1, int ptr)
 {
-	struct record *rec = *(LINE_MAP(index, ptr)->head), *first, *last;
-	get_slice(rec, &first, &last, line1, count1);
-	return first ? first->cnt - (last ? last->cnt : 0) : 0;
+	struct record *rec_ptr, *rec, *last;
+	rec_ptr = LINE_MAP(index, ptr);
+	rec = *(rec_ptr->head);
+	while (rec) {
+		if (line1 <= rec->ptr) {
+			if (rec->ptr > LINE_END(1))
+				break;
+			last = rec->next;
+		}
+		if (!rec->next)
+			break;
+		rec = rec->next;
+	}
+	rec = rec_ptr;
+	return rec ? rec->cnt - (last ? last->cnt : 0) : 0;
 }
 
 static int try_lcs(struct histindex *index, struct region *lcs, int b_ptr,
-- 
1.7.6.12.g6486a.dirty

