From cee890cea949fda56e04cdfe50357b06f12fdf24 Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Fri, 1 Jul 2011 15:20:42 +0800
Subject: [PATCH 26/49] remove old ref code

---
 xdiff/xhistogram.c |    6 ------
 1 files changed, 0 insertions(+), 6 deletions(-)

diff --git a/xdiff/xhistogram.c b/xdiff/xhistogram.c
index f7ba4eb..7b4d763 100644
--- a/xdiff/xhistogram.c
+++ b/xdiff/xhistogram.c
@@ -89,15 +89,11 @@ static int scanA(struct histindex *index, int line1, int count1)
 				new_rec->next = rec;
 				rec->previous = new_rec;
 				*rec_chain = new_rec;
-/* 				index->recs[rec_idx] = REC_CREATE(REC_NEXT(rec), ptr, new_cnt);  */
 				INDEX_NEXT(index, ptr).ptr = rec->ptr;
 				INDEX_NEXT(index, ptr).rec = new_rec;
-/* 				INDEX_NEXT(index, ptr) = rec->ptr;  */
-/* 				INDEX_REC_IDXS(index, ptr) = rec;  */
 				goto continue_scan;
 			}
 
-/* 			rec_idx = REC_NEXT(rec);  */
 			rec = rec->next;
 			chain_len++;
 		}
@@ -115,8 +111,6 @@ static int scanA(struct histindex *index, int line1, int count1)
 		INDEX_NEXT(index, ptr).ptr = 0;
 		INDEX_NEXT(index, ptr).rec = new_rec;
 		*rec_chain = new_rec;
-/* 		INDEX_REC_IDXS(index, ptr) = rec;  */
-/* 		index->records[tbl_idx] = rec_idx;  */
 
 continue_scan:
 		; /* no op */
-- 
1.7.6.12.g6486a.dirty

