From e32efea0ced7b1c781e559bb8fc95102ca4a9baf Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Fri, 1 Jul 2011 20:37:57 +0800
Subject: [PATCH 33/49] remove unused member, ptr from record_next

---
 xdiff/xhistogram.c |    3 ---
 1 files changed, 0 insertions(+), 3 deletions(-)

diff --git a/xdiff/xhistogram.c b/xdiff/xhistogram.c
index c945350..9dc7fc2 100644
--- a/xdiff/xhistogram.c
+++ b/xdiff/xhistogram.c
@@ -17,7 +17,6 @@ struct histindex {
 	unsigned int records_size;
 
 	struct record_next {
-		unsigned int ptr;
 		struct record *rec;
 	} *records_next;
 	unsigned int records_next_size;
@@ -88,7 +87,6 @@ static int scanA(struct histindex *index, int line1, int count1)
 				new_rec->next = rec;
 				rec->previous = new_rec;
 				*rec_chain = new_rec;
-				INDEX_NEXT(index, ptr).ptr = rec->ptr;
 				INDEX_NEXT(index, ptr).rec = new_rec;
 				goto continue_scan;
 			}
@@ -107,7 +105,6 @@ static int scanA(struct histindex *index, int line1, int count1)
 		new_rec->ptr = ptr;
 		new_rec->cnt = 1;
 		new_rec->next = new_rec->previous = NULL;
-		INDEX_NEXT(index, ptr).ptr = 0;
 		INDEX_NEXT(index, ptr).rec = new_rec;
 		*rec_chain = new_rec;
 
-- 
1.7.6.12.g6486a.dirty

