From 7085a8ecbdb6c438c413aed7d8ce5e984d9d5b3d Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Sun, 31 Jul 2011 23:52:57 +0800
Subject: [PATCH 088/130] wip

---
 guilt/gsoc-diff/trim-ends |   90 +++++++++++++++++++++++++++++++++++++++++++++
 1 files changed, 90 insertions(+), 0 deletions(-)

diff --git a/guilt/gsoc-diff/trim-ends b/guilt/gsoc-diff/trim-ends
index e69de29..ad9df13 100644
--- a/guilt/gsoc-diff/trim-ends
+++ b/guilt/gsoc-diff/trim-ends
@@ -0,0 +1,90 @@
+diff --git a/xdiff/xprepare.c b/xdiff/xprepare.c
+index 3ebad0f..1aec64b 100644
+--- a/xdiff/xprepare.c
++++ b/xdiff/xprepare.c
+@@ -130,6 +130,36 @@ static int xdl_classify_record(xdlclassifier_t *cf, xrecord_t **rhash, unsigned
+ }
+ 
+ 
++/*
++ * Early trim initial and terminal matching records.
++ */
++static int xdl_trim_ends(mmfile_t *mf1, mmfile_t *mf2, xpparam_t const *xpp,
++	xdfile_t *xdf1, xdfile_t *xdf2) {
++	long i, lim;
++	xrecord_t **recs1, **recs2;
++
++	recs1 = xdf1->recs;
++	recs2 = xdf2->recs;
++	for (i = 0, lim = XDL_MIN(xdf1->nrec, xdf2->nrec); i < lim;
++	     i++, recs1++, recs2++)
++		if ((*recs1)->ha != (*recs2)->ha)
++			break;
++
++	xdf1->dstart = xdf2->dstart = i;
++
++	recs1 = xdf1->recs + xdf1->nrec - 1;
++	recs2 = xdf2->recs + xdf2->nrec - 1;
++	for (lim -= i, i = 0; i < lim; i++, recs1--, recs2--)
++		if ((*recs1)->ha != (*recs2)->ha)
++			break;
++
++	xdf1->dend = xdf1->nrec - i - 1;
++	xdf2->dend = xdf2->nrec - i - 1;
++
++	return 0;
++}
++
++
+ static int xdl_prepare_ctx(mmfile_t *mf, long narec, xpparam_t const *xpp,
+ 			   xdlclassifier_t *cf, xdfile_t *xdf) {
+ 	unsigned int hbits;
+@@ -239,6 +269,12 @@ int xdl_prepare_env(mmfile_t *mf1, mmfile_t *mf2, xpparam_t const *xpp,
+ 
+ 		return -1;
+ 	}
++	
++	if (xdl_trim_ends(mf1, mf2, xpp, &xe->xdf1, &xe->xdf2) < 0) {
++
++		xdl_free_classifier(&cf);
++		return -1;
++	}
+ 
+ 	if (xdl_prepare_ctx(mf1, enl1, xpp, &cf, &xe->xdf1) < 0) {
+ 
+@@ -403,35 +439,6 @@ static int xdl_cleanup_records(xdfile_t *xdf1, xdfile_t *xdf2) {
+ }
+ 
+ 
+-/*
+- * Early trim initial and terminal matching records.
+- */
+-static int xdl_trim_ends(xdfile_t *xdf1, xdfile_t *xdf2) {
+-	long i, lim;
+-	xrecord_t **recs1, **recs2;
+-
+-	recs1 = xdf1->recs;
+-	recs2 = xdf2->recs;
+-	for (i = 0, lim = XDL_MIN(xdf1->nrec, xdf2->nrec); i < lim;
+-	     i++, recs1++, recs2++)
+-		if ((*recs1)->ha != (*recs2)->ha)
+-			break;
+-
+-	xdf1->dstart = xdf2->dstart = i;
+-
+-	recs1 = xdf1->recs + xdf1->nrec - 1;
+-	recs2 = xdf2->recs + xdf2->nrec - 1;
+-	for (lim -= i, i = 0; i < lim; i++, recs1--, recs2--)
+-		if ((*recs1)->ha != (*recs2)->ha)
+-			break;
+-
+-	xdf1->dend = xdf1->nrec - i - 1;
+-	xdf2->dend = xdf2->nrec - i - 1;
+-
+-	return 0;
+-}
+-
+-
+ static int xdl_optimize_ctxs(xdfile_t *xdf1, xdfile_t *xdf2) {
+ 
+ 	if (xdl_trim_ends(xdf1, xdf2) < 0 ||
-- 
1.7.4.msysgit.0

