From c229ef7d81d1d78d0f44d51d9e0bae35bdbd2164 Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Mon, 1 Aug 2011 00:07:24 +0800
Subject: [PATCH 118/130] remove unused variable

---
 guilt/gsoc-diff/trim-ends |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/guilt/gsoc-diff/trim-ends b/guilt/gsoc-diff/trim-ends
index cc55d31..277dd32 100644
--- a/guilt/gsoc-diff/trim-ends
+++ b/guilt/gsoc-diff/trim-ends
@@ -1,5 +1,5 @@
 diff --git a/xdiff/xprepare.c b/xdiff/xprepare.c
-index 3ebad0f..2548fc2 100644
+index 3ebad0f..d8574f0 100644
 --- a/xdiff/xprepare.c
 +++ b/xdiff/xprepare.c
 @@ -26,6 +26,7 @@
@@ -153,7 +153,7 @@ index 3ebad0f..2548fc2 100644
 +static void xdl_set_dend(xdfile_t *xdf, xpparam_t const *xpp,
 +			 xdlclassifier_t *cf, long lim, long ntail)
 +{
-+	long i, ha;
++	long i;
 +	xrecord_t *rec;
 +
 +	xdf->dend = xdf->nrec - XDL_MIN(lim - xdf->dstart, ntail) - 1;
-- 
1.7.4.msysgit.0

