From 6f79c2cdb9a524b9be82235d97de926228b1090c Mon Sep 17 00:00:00 2001
From: Tay Ray Chuan <rctay89@gmail.com>
Date: Wed, 29 Jun 2011 00:39:01 +0800
Subject: [PATCH 07/15] remove debug function

---
 xdiff/xhistogram.c |   10 ----------
 1 files changed, 0 insertions(+), 10 deletions(-)

diff --git a/xdiff/xhistogram.c b/xdiff/xhistogram.c
index 48e68d3..a91ad38 100644
--- a/xdiff/xhistogram.c
+++ b/xdiff/xhistogram.c
@@ -89,16 +89,6 @@ static unsigned int table_hash(struct histindex *index, int side, int line)
 	return xdl_table_key((map_side(index, side))->recs[line - 1]->ha, index->key_shift);
 }
 
-static void print_record(struct histindex *index, int side, int line)
-{
-	xrecord_t *rec = (map_side(index, side))->recs[line - 1];
-	char *str;
-	str = xdl_malloc(rec->size+1);
-	strncpy(str, rec->ptr, rec->size);
-	str[rec->size+1] = '\0';
-	free(str);
-}
-
 static int scanA(struct histindex *index, int line1, int count1)
 {
 	int ptr, rec_idx, tbl_idx;
-- 
1.7.6.12.g6486a.dirty

